test/e2e/gateway_single_spoke_test.go

```diff
@@ -398,31 +417,28 @@ var _ = Describe("Gateway single target cluster", func() {
 						}).WithContext(ctx).WithTimeout(180 * time.Second).WithPolling(2 * time.Second).ShouldNot(HaveOccurred())
 					}
 
-					By("checking a wildcard cert is present via get request")
+					By("checking tls certificate")
 					{
-						dialer := &net.Dialer{Resolver: authoritativeResolver}
-						dialContext := func(ctx context.Context, network, addr string) (net.Conn, error) {
-							return dialer.DialContext(ctx, network, addr)
-						}
-						http.DefaultTransport.(*http.Transport).DialContext = dialContext
-						http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
-						otherHostname = gatewayapi.Hostname(strings.Join([]string{"other", tconfig.ManagedZone()}, "."))
-						var resp *http.Response
+						certList := &certmanv1.CertificateList{}
 						Eventually(func(ctx SpecContext) error {
-							httpClient := &http.Client{}
-							resp, err = httpClient.Get("https://" + string(otherHostname))
+							err = tconfig.HubClient().List(ctx, certList)
 							if err != nil {
-								GinkgoWriter.Printf("[debug] GET error: '%s'\n", err)
 								return err
 							}
-							err = TestCertificate(string(wildcardHostname), resp)
-							if err != nil {
-								GinkgoWriter.Printf("[debug] Cert error: '%s'\n", err)
-								return err
+							if len(certList.Items) == 0 {
+								return fmt.Errorf("no certificate found")
 							}
-							return nil
-						}).WithTimeout(600 * time.Second).WithPolling(10 * time.Second).WithContext(ctx).ShouldNot(HaveOccurred())
-						defer resp.Body.Close()
+							for _, cert := range certList.Items {
+								if cert.Labels["gateway"] == testID {
+									for _, dnsName := range cert.Spec.DNSNames {
+										if dnsName == string(wildcardHostname) {
+											return nil
+										}
+									}
+								}
+							}
+							return fmt.Errorf("dns names for certificate not as expected")
+						}).WithContext(ctx).WithTimeout(180 * time.Second).WithPolling(2 * time.Second).ShouldNot(HaveOccurred())
 					}
```

Comment:
Lately I've been trying to better use the gomega matchers in these tests, what about trying somethign like this:

```.go
By("checking tls certificate")
{
	Eventually(func(g Gomega, ctx context.Context, expected ...string) {
		certList := &certmanv1.CertificateList{}
		err := tconfig.HubClient().List(ctx, certList)
		g.Expect(err).NotTo(HaveOccurred())
		g.Expect(certList.Items).To(Not(BeEmpty()))
		g.Expect(certList.Items).Should(ContainElement(
			MatchFields(IgnoreExtras, Fields{
				"ObjectMeta": MatchFields(IgnoreExtras, Fields{
					"Labels": HaveKeyWithValue("gateway", testID),
				}),
				"Spec": MatchFields(IgnoreExtras, Fields{
					"DNSNames": ConsistOf(string(hostname), string(wildcardHostname)),
				}),
			})))
	}).WithContext(ctx).Should(Succeed())
}
```